lib/spack/spack/variant.py

```diff
@@ -367,13 +401,13 @@ def satisfies(self, other):
         Returns:
             bool: True or False
         """
-        # If names are different then `self` does not satisfy `other`
-        # (`foo=bar` does not satisfy `baz=bar`)
-        if other.name != self.name:
-            return False
+        # If the superclass doesn't satisfy the superclass, it doesn't satisfy
+        # this handles conflicts between none and any
```

Comment:
I'm having trouble parsing this comment.

Also, I think this (and SingleValuedVariant) would benefit if the 'any'/None testing were added as a static method on AbstractVariant and called directly (rather than accessing this via superclass).

Reply:
```
if not super(a).satisfies(super(b)):
    not a.satisfies(b)
```

Reply:
Handling this by fixing the class hierarchy instead.